tutorials/dataframe/df022_createKahan.C

```diff
@@ -0,0 +1,98 @@
+/// \file
+/// \ingroup tutorial_dataframe
+/// \notebook
+/// This tutorial shows how to implement a Kahan summation custom action.
+///
+/// \macro_code
+///
+/// \date July 2018
+/// \author Enrico Guiraud, Danilo Piparo, Massimo Tumolo
+
+class KahanSum : public ROOT::Detail::RDF::RActionImpl<class KahanSum> {
+public:
+   /// This type is a requirement for every helper.
+   using Result_t = double;
+
+private:
+   std::vector<double> fPartialSums;
+   std::vector<double> fCompensations;
+   int fNSlots;
+
+   std::shared_ptr<double> fResultSum;
+
+public:
+   KahanSum(KahanSum &&) = default;
+   KahanSum(const KahanSum &) = delete;
+
+   KahanSum(const std::shared_ptr<double> &r) : fResultSum(r)
+   {
+      fNSlots = ROOT::IsImplicitMTEnabled() ? ROOT::GetImplicitMTPoolSize() : 1;
+      fPartialSums.resize(fNSlots, 0.0);
+      fCompensations.resize(fNSlots, 0.0);
+   }
+
+   std::shared_ptr<Result_t> GetResultPtr() const { return fResultSum; }
+
+   void Initialize() {}
+   void InitTask(TTreeReader *, unsigned int) {}
+
+   template <typename T>
+   void Exec(unsigned int slot, T x)
+   {
+      double y = x - fCompensations[slot];
```

Comment:
Would it make sense to extract the Kahan summation algo in a private method and use it both here and in Finalize?